backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/DiagramContext.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.spring.collaborative.diagrams;
+
+import java.util.Objects;
+
+import org.eclipse.sirius.web.collaborative.diagrams.api.IDiagramContext;
+import org.eclipse.sirius.web.diagrams.Diagram;
+
+/**
+ * The implementation of {@link IDiagramContext}.
+ *
+ * @author hmarchadour
+ */
+public class DiagramContext implements IDiagramContext {
```

Comment:
This feels like reinventing something like `AtomicReference` (without the thread safety).